lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -615,54 +546,81 @@ void FMemModuleOp::insertPorts(ArrayRef<std::pair<unsigned, PortInfo>> ports) {
     return;
 
   // Drop the direction markers for dead ports.
   SmallVector<Direction> portDirections =
-      direction::unpackAttribute(this->getPortDirectionsAttr());
-  ArrayRef<Attribute> portNames = this->getPortNames();
-  ArrayRef<Attribute> portTypes = this->getPortTypes();
-  ArrayRef<Attribute> portAnnos = this->getPortAnnotations();
-  ArrayRef<Attribute> portSyms = this->getPortSymbols();
-  assert(portDirections.size() == getNumPorts());
-  assert(portNames.size() == getNumPorts());
-  assert(portAnnos.size() == getNumPorts() || portAnnos.empty());
-  assert(portTypes.size() == getNumPorts());
-  assert(portSyms.size() == getNumPorts() || portSyms.empty());
+      direction::unpackAttribute(op.getPortDirectionsAttr());
+  ArrayRef<Attribute> portNames = op.getPortNames();
+  ArrayRef<Attribute> portTypes = op.getPortTypes();
+  ArrayRef<Attribute> portAnnos = op.getPortAnnotations();
+  ArrayRef<Attribute> portSyms = op.getPortSymbols();
+  assert(portDirections.size() == op.getNumPorts());
+  assert(portNames.size() == op.getNumPorts());
+  assert(portAnnos.size() == op.getNumPorts() || portAnnos.empty());
+  assert(portTypes.size() == op.getNumPorts());
+  assert(portSyms.size() == op.getNumPorts() || portSyms.empty());
 
   SmallVector<Direction> newPortDirections =
       removeElementsAtIndices<Direction>(portDirections, portIndices);
-  SmallVector<Attribute> newPortNames =
-      removeElementsAtIndices(portNames, portIndices);
-  SmallVector<Attribute> newPortTypes =
-      removeElementsAtIndices(portTypes, portIndices);
-  SmallVector<Attribute> newPortAnnos =
-      removeElementsAtIndices(portAnnos, portIndices);
-  SmallVector<Attribute> newPortSyms =
-      removeElementsAtIndices(portSyms, portIndices);
-  (*this)->setAttr("portDirections",
-                   direction::packAttribute(getContext(), newPortDirections));
-  (*this)->setAttr("portNames", ArrayAttr::get(getContext(), newPortNames));
-  (*this)->setAttr("portAnnotations",
-                   ArrayAttr::get(getContext(), newPortAnnos));
-  (*this)->setAttr("portTypes", ArrayAttr::get(getContext(), newPortTypes));
-  (*this)->setAttr("portSyms", ArrayAttr::get(getContext(), newPortSyms));
+  SmallVector<Attribute> newPortNames, newPortTypes, newPortAnnos, newPortSyms;
+  newPortNames = removeElementsAtIndices(portNames, portIndices);
+  newPortTypes = removeElementsAtIndices(portTypes, portIndices);
+  newPortAnnos = removeElementsAtIndices(portAnnos, portIndices);
+  newPortSyms = removeElementsAtIndices(portSyms, portIndices);
+  op->setAttr("portDirections",
+              direction::packAttribute(op.getContext(), newPortDirections));
+  op->setAttr("portNames", ArrayAttr::get(op.getContext(), newPortNames));
+  op->setAttr("portAnnotations", ArrayAttr::get(op.getContext(), newPortAnnos));
+  op->setAttr("portTypes", ArrayAttr::get(op.getContext(), newPortTypes));
+  op->setAttr("portSyms", ArrayAttr::get(op.getContext(), newPortSyms));
```

Comment:
Should this be checking if we removed all port symbols, and if so replace with empty array?

Reply:
Thanks for spotting this. Addressed it by calling the `fixupPortSymsArray`.